include/picongpu/param/binningSetup.param

```diff
@@ -0,0 +1,211 @@
+/* Copyright 2023 Tapish Narwal
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "picongpu/simulation_defines.hpp"
+
+#include "picongpu/plugins/binning/binnerPlugin.hpp"
+
+namespace picongpu
+{
+    namespace plugins::binning
+    {
+        void binningExample(BinningCreator& binningCreator);
+
+        inline void getBinning(BinningCreator& binningCreator)
+        {
+            /// USER CODE GOES BELOW
+
+            /// USER CODE ENDS HERE
+        }
+
+        template<typename T_Identifier>
+        struct NameOrTypeExists
+        {
+            template<typename T_Value>
+            struct HasTypeOrName
+            {
+                static constexpr bool value
+                    = std::is_same_v<
+                          T_Identifier,
+                          T_Value> || std::is_same_v<pmacc::traits::GetCTName_t<T_Value>, T_Identifier>;
+            };
+
+            using FilteredSeq = mp_copy_if<VectorAllSpecies, HasTypeOrName>;
+
+            static constexpr bool value = !std::is_same_v<boost::mp11::mp_true, mp_empty<FilteredSeq>>;
+        };
+
+        inline void binningExample(BinningCreator& binningCreator)
```

Comment:
please add a comment that this is the check that the species exists. Best use

```
constexpr speciesExists = ....;
if constexpr(speciesExists) ...
```

Reply:
Comment added